### PR TITLE
[MM-34838] Remove Ancillary Permissions That Are Purposefully Being Removed

### PR DESCRIPTION
**Pull request by hahmadia**
#### Summary

If I have a section permission that gives me ancillary permissions, I cannot remove an ancillary permission individually via MMCTL anymore. This PR aims to resolve this issue.

For example. 
I have SYSCONSOLE_XXXX_WRITE which gives me `manage_team` permission, if I try to remove `manage_team`, it will always keep being re-added.

Discussion started here
https://github.com/mattermost/mattermost-server/pull/16917#discussion_r612743973

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-34838

#### Release Note

```release-note
None
```

### CODE REVIEW

**Comment by hahmadia** [comment]
Please note that tests are on the way. I've just added you guys to get a look ahead of time.

File: model/role.go [hunk 1]
```diff
@@ -950,3 +950,43 @@ func AddAncillaryPermissions(permissions []string) []string {
 	}
 	return permissions
 }
+
+// This function removes any ancillary permissions that user has removed from their new Permissions without having removed the sub-section permission
```
[the remark this answers is not in the document]
Comment (mkraft) [reply model/role.go:954]: Also, this isn't really *removing* ancillary permissions. It's doing some set operations. As far as I can tell it's doing...

* Let *O* be the old permissions.
* Let *N* be the new permissions.
* Let *A* be the subset of *N* that are ancillary.

`RemoveUnwantedAncillaryPermissions` = (*N* - *O*) ⋃ *A*

Can this be simplified to: *N* - *A*?

I may be totally wrong. Let me know.

**Review by mkraft — CHANGES_REQUESTED** [review]

**Comment by amyblais** [comment]
@hahmadia v5.35 will be based on Cloud April 21st release which is coming on Wednesday. Is this and the other bug fixes still on track?

**Pull request closed without merging**